pennylane/operation.py

```diff
@@ -972,7 +977,7 @@ def terms(self) -> tuple[list[TensorLike], list["Operation"]]:  # pylint: disabl
         """
         raise TermsUndefinedError
 
-    num_wires: Union[int, WiresEnum] = AnyWires
+    num_wires: Optional[Union[int, WiresEnum]] = None
```

Comment:
Maybe worth removing `WiresEnum` from the type hint as part of the deprecation?

Reply:
Technically it's still allowed to be a `WiresEnum` and will work with it.